HW_1/hw_1.py

```diff
@@ -0,0 +1,466 @@
++ `DNASequence` должен иметь метод `transcribe`, возвращающий транскрибированную РНК-последовательность
++ Данные классы не должны иметь <ins>публичных методов</ins> `complement` и метода для проверки алфавита, так как они уже должны быть реализованы в `NucleicAcidSequence`.
+
+Напишите класс `AminoAcidSequence`:
++ Данный класс реализует интерфейс `BiologicalSequence`
++ Добавьте этому классу один любой метод, подходящий по смыслу к аминокислотной последовательности. Например, метод для нахождения изоэлектрической точки, молекулярного веса и т.д.
+
+Комментарий по поводу метода `NucleicAcidSequence.complement`, так как я хочу, чтобы вы сделали его опредедённым образом:
+
+При вызове `dna.complement()` или условного `dna.check_alphabet()` должны будут вызываться соответствующие методы из `NucleicAcidSequence`. При этом, данный метод должен обладать свойством полиморфизма, иначе говоря, внутри `complement` не надо делать условия а-ля `if seuqence_type == "DNA": return self.complement_dna()`, это крайне не гибко. Данный метод должен опираться на какой-то общий интерфейс между ДНК и РНК. Создание экземпляров `NucleicAcidSequence` не подразумевается, поэтому код `NucleicAcidSequence("ATGC").complement()` не обязан работать, а в идеале должен кидать исключение `NotImplementedError` при вызове от экземпляра `NucleicAcidSequence`
+
+Вся сложность задания в том, чтобы правильно организовать код. Если у вас есть повторяющийся код в сестринских классах или родительском и дочернем, значит вы что-то делаете не так.
+
+
+Маленькое замечание: По-хорошему, между классом `BiologicalSequence` и классами `NucleicAcidSequence` и `AminoAcidSequence`, ещё должен быть класс-прослойка, частично реализующий интерфейс `BiologicalSequence`, но его писать не обязательно, так как задание и так довольно большое (правда из-за этого у вас неминуемо возникнет повторяющийся код в классах `NucleicAcidSequence` и `AminoAcidSequence`)
+"""
+
+from abc import ABC, abstractmethod
+
+class BiologicalSequence(ABC):    
+    @abstractmethod
+    def __len__(self):
+        pass
+    
+    @abstractmethod
+    def __getitem__(self, ind):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+    
+    @abstractmethod
+    def __str__(self):
+        pass
+    
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+
+    dna_complement = {"A": "T", "a": "t",
+                      "T": "A", "t": "a",
+                      "G": "C", "g": "c",
+                      "C": "G", "c": "g"}
+
+    rna_complement = {"A": "U", "a": "u",
+                      "U": "A", "u": "a",
+                      "G": "C", "g": "c",
+                      "C": "G", "c": "g"}
+
+    def __init__(self, seq:str):
+        self.seq = seq
+    
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, ind):
+        return self.seq.__getitem__(ind)
```

Comment:
Если можно не использовать дандер, то не стоит использовать дандер
```suggestion
        return self.seq[ind]
```